### PR TITLE
Add native GCS support for goofys

### PR DESCRIPTION
**Pull request by dotslash**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
 go 1.14
 
 require (
-	github.com/Azure/azure-pipeline-go v0.2.2
-	github.com/Azure/azure-sdk-for-go v32.1.0+incompatible
-	github.com/Azure/azure-storage-blob-go v0.7.1-0.20190724222048-33c102d4ffd2
+	cloud.google.com/go/storage v1.14.0
+	github.com/Azure/azure-pipeline-go v0.2.3
```
Comment (dotslash) [inline go.mod:7]:
```suggestion
	github.com/Azure/azure-pipeline-go v0.2.2
```

File: go.mod [hunk 2]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
+	github.com/Azure/azure-sdk-for-go v52.6.0+incompatible
```
Comment (dotslash) [inline go.mod:8]:
```suggestion
	github.com/Azure/azure-sdk-for-go v32.1.0+incompatible
```

File: go.mod [hunk 3]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
+	github.com/Azure/azure-storage-blob-go v0.13.0
```
Comment (dotslash) [inline go.mod:9]:
```suggestion
	github.com/Azure/azure-storage-blob-go v0.7.1-0.20190724222048-33c102d4ffd2
```

File: go.mod [hunk 4]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
 	github.com/Azure/go-autorest/autorest/azure/cli v0.4.2
 	github.com/Azure/go-autorest/autorest/to v0.4.0 // indirect
 	github.com/Azure/go-autorest/autorest/validation v0.3.1 // indirect
-	github.com/aws/aws-sdk-go v1.38.7
-	github.com/google/uuid v1.1.2-0.20190416172445-c2e93f3ae59f
-	github.com/gopherjs/gopherjs v0.0.0-20210202160940-bed99a852dfe // indirect
-	github.com/jacobsa/fuse v0.0.0-20201216155545-e0296dec955f
-	github.com/jtolds/gls v4.2.0+incompatible // indirect
+	github.com/StackExchange/wmi v0.0.0-20210224194228-fe8f1750fd46 // indirect
+	github.com/aws/aws-sdk-go v1.38.10
```
Comment (dotslash) [inline go.mod:17]:
```suggestion
	github.com/aws/aws-sdk-go v1.38.7
```

File: go.mod [hunk 5]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
+	github.com/go-akka/configuration v0.0.0-20200606091224-a002c0330665
+	github.com/go-ole/go-ole v1.2.5 // indirect
+	github.com/gofrs/uuid v4.0.0+incompatible // indirect
+	github.com/google/uuid v1.2.0
```
Comment (dotslash) [inline go.mod:21]:
```suggestion
	github.com/google/uuid v1.1.2-0.20190416172445-c2e93f3ae59f
```

File: go.mod [hunk 6]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
+	github.com/hashicorp/go-retryablehttp v0.6.8
+	github.com/jacobsa/fuse v0.0.0-20210330112455-9677d0392291
```
Comment (dotslash) [inline go.mod:23]:
```suggestion
	github.com/jacobsa/fuse v0.0.0-20201216155545-e0296dec955f
```

File: go.mod [hunk 7]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
 	github.com/kardianos/osext v0.0.0-20190222173326-2bc1f35cddc0
-	github.com/kr/pretty v0.1.1-0.20190720101428-71e7e4993750 // indirect
 	github.com/kylelemons/godebug v1.1.0 // indirect
-	github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb // indirect
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/pkg/errors v0.9.1 // indirect
-	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b
+	github.com/rifflock/lfshook v0.0.0-20180920164130-b9218ef580f5
+	github.com/satori/go.uuid v1.2.0
```
Comment (dotslash) [inline go.mod:28]:
```suggestion
	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b
```

File: go.mod [hunk 8]
```diff
@@ -3,36 +3,42 @@ module github.com/kahing/goofys
...
-	github.com/sirupsen/logrus v1.4.3-0.20190807103436-de736cf91b92
-	github.com/smartystreets/assertions v0.0.0-20160201214316-443d812296a8 // indirect
-	github.com/smartystreets/goconvey v1.6.1-0.20160119221636-995f5b2e021c // indirect
-	github.com/urfave/cli v1.21.1-0.20190807111034-521735b7608a
-	golang.org/x/sys v0.0.0-20200930185726-fdedc70b468f
-	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
-	gopkg.in/ini.v1 v1.46.0
+	github.com/shirou/gopsutil v3.21.2+incompatible
+	github.com/sirupsen/logrus v1.8.1
+	github.com/smartystreets/goconvey v1.6.4 // indirect
```
Comment (dotslash) [inline go.mod:32]:
```suggestion
	github.com/smartystreets/goconvey v1.6.1-0.20160119221636-995f5b2e021c // indirect
```

**Comment by gaul** [comment]
After this merges it would be good to add a comparison against gcsfuse to the README.

**Pull request merged**